src/Config.php

```diff
@@ -69,4 +80,23 @@ public function clear(string $cacheDir): void
     {
         $this->cache->clear();
     }
+
+    /**
+     * @return array<mixed>
+     */
+    private function getConfig(): array
+    {
+        if (is_array($this->config)) {
```

Comment:
I don't think the is_array check is needed. You can just do: if($this->config)

Reply:
The config is possibly `null`. If it is `null`, we should compile it.
This add some "caching" on the same request, so it doesn't compile each time it fetches the cache.

Reply:
yes, but what I mean was: you don't need to check this is an array because it can only be null or an array. so, if ($this->config) would suffice. But it's a minor remark